### PR TITLE
VFB 66 - onSwapRows amended to directly update the row order

### PR DESCRIPTION
**Pull request by mtharkins**
## What's changed
The login of onSwapRows has been updated to use the listOfIngridients function as props to update the row_order value of listsData

## Checklist
- [x] The ticket is up-to-date - Please document any deviations from the original approach if there is any.
- [x] I have documented the testing steps for QA 
- [x] Make sure you've verified it works via `npm run dev`
- [x] Make sure you've verified it works via `npm run build` and `npm run start`
- [x] Make sure you've fixed all linting problems with `npm run lint_fix`
  - Will also be automatically checked on pull request
- [x] Make sure you've tested via `npm run test`
  - Will also be automatically checked on pull request

### CODE REVIEW

File: src/app/lists/ListDataview.tsx [hunk 1]
```diff
@@ -36,8 +36,9 @@ interface QuantityAndNotes {
     notes: string | null;
 }
 
-interface Props {
-    data: Schema["lists"][];
+interface ListDataViewProps {
+    listOfIngridients: Schema["lists"][];
+    setData: React.Dispatch<React.SetStateAction<Schema["lists"][]>>;
```
Comment (rickaw) [inline src/app/lists/ListDataview.tsx:41]: should we also rename this, like `setListOfIngredients`?

File: src/app/lists/ListDataview.tsx [hunk 2]
```diff
@@ -133,9 +134,28 @@ const ListsDataView: React.FC<Props> = (props) => {
...
+
+        item1.row_order = order2;
+        item2.row_order = order1;
+
+        const newData = [...listOfIngridients];
+
+        newData[index1] = item2;
+        newData[index2] = item1;
+
+        setData(newData);
```
Comment (rickaw) [inline src/app/lists/ListDataview.tsx:158]: maybe worth putting this in a separate function as discussed :)

**Review by rickaw — COMMENTED** [review]
Nice work! a few comments

File: src/app/lists/ListDataview.tsx (hunk 1 above)
Comment (mtharkins) [reply src/app/lists/ListDataview.tsx:41]: Renamed

File: src/app/lists/ListDataview.tsx (hunk 2 above)
Comment (mtharkins) [reply src/app/lists/ListDataview.tsx]: Moved this into a separate function

File: src/app/lists/ListDataview.tsx [hunk 3]
```diff
@@ -80,18 +81,22 @@ const listsColumnStyleOptions: ColumnStyles<ListRow> = {
     ),
 };
 
-const ListsDataView: React.FC<Props> = (props) => {
+const ListsDataView: React.FC<ListDataViewProps> = ({
+    listOfIngridients,
+    setListOfIngridients: setData,
```
Comment (rickaw) [inline src/app/lists/ListDataview.tsx:86]: oops, seems like the same thing has happened here too. If we could just use `setListOfIngridients` throughout, that'd be fab

File: src/app/lists/ListDataview.tsx [hunk 4]
```diff
@@ -114,9 +119,33 @@ const ListsDataView: React.FC<Props> = (props) => {
...
+        const row1Item = listOfIngridients[row1Index];
+        const row1Order = row1Item.row_order;
+
+        const row2Item = listOfIngridients[row2Index];
+        const row2Order = row2Item.row_order;
+
+        row1Item.row_order = row2Order;
+        row2Item.row_order = row1Order;
+
+        const newRowOrder = [...listOfIngridients];
```
Comment (rickaw) [inline src/app/lists/ListDataview.tsx:142]: shall we rename this to `newListOfIngredients`?

**Review by rickaw — APPROVED** [review]
Looks great! A few minor comments but feel free to merge once they're addressed

**Pull request merged**